pandas/core/frame.py

```diff
@@ -8281,10 +8281,10 @@ def _get_data(axis_matters):
             out_dtype = "bool" if filter_type == "bool" else None
 
             def blk_func(values):
-                if values.ndim == 1 and not isinstance(values, np.ndarray):
-                    # we can't pass axis=1
-                    return op(values, axis=0, skipna=skipna, **kwds)
-                return op(values, axis=1, skipna=skipna, **kwds)
+                if isinstance(values, ExtensionArray):
+                    return values._reduce(name, skipna=skipna, **kwds)
```

Comment:
is _reduce part of the interface?

if we're avoiding nanops, can/should we make those functions either a) not handle EA cases or b) dispatch to _reduce?